src/java/org/apache/cassandra/db/monitoring/MonitoringTask.java

```diff
@@ -328,31 +379,126 @@ protected abstract static class Operation
          * this is set lazily as it takes time to build the query CQL */
         private String name;
 
+        /**
+         * creation time of this Operation object, in ms,
+         * this is different from operation's creationTimeNanos
+         * which does not follow wall clock and is useless for
+         * reporting purposes e.g. in virtual tables
+         */
+        private final long timestamp;
```

Comment:
can we add time unit to the variable name, like timestampMs?
it is easy to mix timestamp with different time units (like in seconds vs microseconds)